docs/chain/fullnode.md

```diff
@@ -0,0 +1,107 @@
+:::
+
+### WASM runtime execution
+
+A KILT full node should use the `--execution=wasm` parameter for both the relaychain and parachain collation.
+The alternative to WASM runtime execution is native runtime execution, which might be faster but can, in some cases, deviate from the WASM execution logic and result in a different state.
+When this happens the full node will crash and will stop producing blocks.
+Since the WASM runtime logic is part of the blockchain state itself and hence represents the single source of truth, all nodes should execute the WASM version of the runtime logic.
+
+### Specify the right chain spec
+
+The `--chain` parameter decides which blockchain the KILT full node will join.
+This parameter must be specified for both the parachain and the relaychain, since both chains are, as a matter of fact, separate blockchains.
+The KILT parachain accepts an additional parameter to select the environment to use for the WASM runtime execution.
+This can either be `peregrine` or `spiritnet`.
+
+Hence, to start a full node for the spiritnet network, the parameter would be `--chain=spiritnet`, while for peregrine would be `--chain=peregrine`.
+
+### Where are all the files stored?
+
+The `--base-path` parameter specifies where all the persistent files must be stored.
+By default the session keys will also be stored in the _base path_, but we recommend to separate them from the other files.
+This makes sure that the keyfiles are not accidentally lost or published when the blockchain database is backed up or restored.
+
+## Running an Archive Node
+
+We recommend following the instructions in the KILT chain repository. Below is the command to build the KILT full node executable. The command must be run from the root directory of the repository after it has been cloned.
+
+```bash
+# Set up the build environment by installing the Rust compiler.
+./scripts/init.sh
+# Build the executable from source enabling all the optimisations with --release.
+cargo build --release -p kilt-parachain
+```
+
+:::info
+
+We discourage to use the `develop` branch to build the executable. Instead, the latest commit from `master` should be used.
+
+:::
+
+The compiled executable can be found in `./target/release/kilt-parachain` after the build process completes successfully.
+
+```bash
+./target/release/kilt-parachain \
+  --chain={spiritnet, peregrine} \
+  --runtime={spiritnet, peregrine} \
+  --rpc-port=9933 \
+  --rpc-cors=all \
+  --rpc-external \
+  --ws-external \
+  --name="name of full node" \
+  --execution=wasm \
+  --pruning archive \
+  -- \
+  --chain=kusama \
+  --execution=wasm
+```
+
+## Sync the Blockchain State
+
+The node needs to fully sync up with both the parachain and the relaychain.
+Depending on the size of the blockchain states, it may take a number of hours to few days for the node to catch up.
```

Comment:
```suggestion
Depending on the size of the blockchain state and your hardware, it may take a number of hours to few days for the node to catch up.
```